SingularityService/src/main/java/com/hubspot/singularity/resources/TaskResource.java

```diff
@@ -180,6 +183,24 @@ public SingularityTaskRequest getPendingTask(@Auth SingularityUser user, @PathPa
     return taskRequestList.get(0);
   }
 
+  @DELETE
+  @Path("/scheduled/task/{scheduledTaskId}")
+  @ApiOperation("Delete a scheduled task.")
+  public Void deleteScheduledTask(@Auth SingularityUser user, @PathParam("scheduledTaskId") String taskId, @Context HttpServletRequest requestContext) {
+    return maybeProxyToLeader(requestContext, Void.class, null, () -> deleteScheduledTask(taskId, user));
+  }
+
+  private Void deleteScheduledTask(String taskId, SingularityUser user) {
```

Comment:
Two things here:
- We should keep this public instead of private. These methods are called when the endpoint is hit, so I would generally consider them publicly accessible. It's also useful to be able to test the methods for endpoints end to end (i.e. with the validation/etc) in other parts of the code when needed
- Rather than returning a void, it might be useful to return an actual object. If you look in other DELETE methods in Singularity we generally return some sort of 'receipt' for the delete. For example, killing a task returns the task cleanup object that was created. Deleting a request returns the request data that was deleted. For this case maybe we can return the SingularityPendingTaskId that is being deleted?